src/midi/midi_mt32.cpp

```diff
@@ -61,12 +65,100 @@ MidiHandler_mt32 mt32_instance;
 static void init_mt32_dosbox_settings(Section_prop &sec_prop)
 {
 	constexpr auto when_idle = Property::Changeable::WhenIdle;
-	auto *str_prop = sec_prop.Add_string("romdir", when_idle, "");
+
+	const char *models[] = {"auto", "cm32l", "mt32", 0};
+	auto *str_prop = sec_prop.Add_string("model", when_idle, "auto");
+	str_prop->Set_values(models);
+	str_prop->Set_help(
+	        "Model of synthesizer to use. The default (auto) prefers CM-32L\n"
+	        "if both sets of ROMs are provided. For early Sierra games and Dune 2\n"
+	        "it's recommended to use 'mt32', while newer games typically made\n"
+	        "use of the CM-32L's extra sound effects (use 'auto' or 'cm32l')");
+
+	str_prop = sec_prop.Add_string("romdir", when_idle, "");
 	str_prop->Set_help(
-	        "The directory holding the required MT-32 Control and PCM ROMs.\n"
-	        "The ROM files should be named as follows:\n"
-	        "  MT32_CONTROL.ROM or CM32L_CONTROL.ROM - control ROM file.\n"
-	        "  MT32_PCM.ROM or CM32L_PCM.ROM - PCM ROM file.");
+	        "The directory holding the required MT-32 and/or CM-32L ROMs\n"
+	        "named as follows:\n"
+	        "  MT32_CONTROL.ROM or CM32L_CONTROL.ROM - control ROM files(s).\n"
+	        "  MT32_PCM.ROM or CM32L_PCM.ROM - PCM ROM file(s).\n"
+	        "The directory can be absolute or relative, or leave it blank to\n"
+	        "use the 'mt32-roms' directory in your DOSBox configuration\n"
+	        "directory, followed by checking other common system locations.");
+}
+
+#if defined(WIN32)
+
+static std::list<std::string> get_rom_dirs(const std::string &preferred_dir)
```

Comment:
I really suggest avoiding C++ std::list whenever possible. In some contexts it might make sense, but in here std::vector is better than std::list in every way. Content is tiny though, so it's not a big deal.

Reply:
Indeed.. I noticed the benchmarks here:

https://baptiste-wicht.com/posts/2012/12/cpp-benchmark-vector-list-deque.html

Can you switch it over to vector? (On `master`)

Reply:
Done :)